### PR TITLE
Recipe editing, form validation, notifications and update to use typescript

### PR DESCRIPTION
**Pull request by ErikasLau**
- Added recipe editing which can be accesed from „My Recipes“ page with validation of recipe author;
- Updated login, register, create and edit forms to use form validation tool. Now forms instantly show feedback on the input;
- Added in web app notification support to the website so any page in the website can define a message to show which will be shown in any page;
- updated code base to use typescript.

### CODE REVIEW

File: Receptai.client/pages/user/dashboard/My-recipes/Create/index_copy.vue
Comment (svetikas) [inline Receptai.client/pages/user/dashboard/My-recipes/Create/index_copy.vue:1]: Kokia yra sio failo egzistencijos prasme?

File: docker-compose.yaml
Comment (svetikas) [inline docker-compose.yaml:1]: Priminimas sau: pasalinti visus failo pakeitimus

File: Receptai.client/components/admin/RecipeEdit/RecipeEditingPage_copy.vue
Comment (svetikas) [inline Receptai.client/components/admin/RecipeEdit/RecipeEditingPage_copy.vue:1]: Kokia yra sio failo egzistencine prasme?

File: Receptai.client/components/Notification.vue [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+          ? 'hover:!shadow-sm hover:!shadow-[#d9000085] focus:!ring-red-600'
+          : store.label === 'Success'
+          ? 'hover:!shadow-sm hover:!shadow-emerald-400 focus:!ring-emerald-500'
+          : null
+      "
+      data-dismiss-target="#toast-default"
+      aria-label="Close"
+    >
+      <span class="sr-only">Close</span>
+      <svg
```
Comment (svetikas) [inline Receptai.client/components/Notification.vue]: Reiktu komentaro apie tai, kas yra sitas SVG

**Comment by ErikasLau** [comment]
Updated Receptai.client-dev branch to correct mistakes on security vulnerability, not clear svg name/purpose in notification. Deleted not used files.

**Review by svetikas — APPROVED** [review]
Daugiau didelių problemų nematau, ateičiai rekomenduoju pull request'ą kurti kiekvienam įgyvendintui jira ticketui ar feature, o ne viska vienu metu, nes tai apsunkina kodo apžvalga.

**Pull request merged**